Refactored Names on every Individual instance

vagmark30 opened the pull request:
Closes #69 
Checked post and put methods on Swagger

vagmark30 committed:
Refactored Names on every Individual instance

esentis commented on src/Ieemdb.Persistence/Models/Actor.cs:
```diff
@@ -7,14 +7,26 @@ namespace Esentis.Ieemdb.Persistence.Models
       {
-        name = value;
+        lName = value;
         NormalizedSearch = value.NormalizeSearch();
```
Το NormalizedSearch πριν ήταν μόνο για το FirstName, τώρα που βάζουμε και LastName κανεις normalize και αλλαζεις την ιδια μεταβλητη. Τωρα αναγκαστικά θα έχουμε δύο NormalizedSearch, NormalizedFirstName και NormalizedLastName. Αυτό ισχύει σε όλα τα μοντέλα που έχει το NormalizedSearch.

esentis reviewed (changes requested):
Δές το comment που έβαλα ;)

vagmark30 committed:
Changed normalizedSearch to NormalizedFirstNameSearch, NormalizedLastNameSearch
WIP - Work is required to files IeemdbDbContext.cs  DbContextExtensions.cs for the search function

vagmark30 commented:
Work is required to files IeemdbDbContext.cs  DbContextExtensions.cs for the ability to search function.

esentis commented on src/Ieemdb.Persistence/IeemdbDbContext.cs:
```diff
@@ -55,19 +55,19 @@ protected override void OnModelCreating(ModelBuilder builder)
     {
       base.OnModelCreating(builder);
       builder.Entity<Actor>()
-        .HasIndex(e => e.NormalizedSearch)
+        .HasIndex(e => e.NormalizedLastNameSearch)
         .IsTsVectorExpressionIndex("english");
 
       builder.Entity<Director>()
-        .HasIndex(e => e.NormalizedSearch)
+        .HasIndex(e => e.NormalizedLastNameSearch)
         .IsTsVectorExpressionIndex("english");
 
       builder.Entity<Movie>()
         .HasIndex(e => e.NormalizedSearch)
         .IsTsVectorExpressionIndex("english");
 
       builder.Entity<Writer>()
-        .HasIndex(e => e.NormalizedSearch)
+        .HasIndex(e => e.NormalizedLastNameSearch)
         .IsTsVectorExpressionIndex("english");
 
       builder.Entity<MovieActor>(e =>
```
Πρέπει να φύγουν αυτά, το NormalizedLastNameSearch / NormalizedFirstNameSearch δεν πρέπει να είναι index της βασης. Το NormalizeSearch πρεπει να μείνει γιατι είναι για το Full Text Search. 
Π.χ 
```csharp
 builder.Entity<Actor>()
        .HasIndex(e => e.NormalizedSearch)
        .IsTsVectorExpressionIndex("english");
```

esentis commented on src/Ieemdb.Persistence/Abstractions/ISearchableNames.cs:
```diff
@@ -0,0 +1,15 @@
+namespace Esentis.Ieemdb.Persistence.Abstractions
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Linq;
+  using System.Text;
+  using System.Threading.Tasks;
+
+  public interface ISearchableNames
+  {
+    string NormalizedFirstNameSearch { get; }
+
+    string NormalizedLastNameSearch { get; }
+  }
+}
```
Δεν υπάρχει λόγος να το κάνεις interface, ας ειναι πεδία μεσα στο Model.

esentis reviewed (changes requested):
Αν δείς το DbContext extension το οποιο κάνει στην ουσία το Full Text Search του λεω να κοιτάξει το NormalizedSearch πεδίο του Entity που ψάχνω.
 ```csharp
 public static class DbContextExtensions
  {
    public static IQueryable<TEntity> FullTextSearch<TEntity>(this IQueryable<TEntity> source, string query)
      where TEntity : ISearchable
      => source
        .Where(x => EF.Functions.ToTsVector("english", x.NormalizedSearch).Matches(query));
  }
  ```
  
  Επειδή Full Text Search γίνεται σε φυσική γλώσσα, το normalizedSearch στους Actors/Writers/Directors είναι το Bio τους (δηλαδή θα κάνει ότι έκανες με το άλλο πεδίο απλά θα ειναι το Bio που πρέπει να περασει normalization), και στις ταινίες θα είναι ο τίτλος και γιατι όχι η πλοκή.

vagmark30 committed:
Fixed normalized names, added normalized search field on bio on each individual

esentis reviewed (approved):
Νομίζω πως είμαστε καλα για τωρα :P
